hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/services/DefinitionsService.java

```diff
@@ -37,31 +37,132 @@
     /*
         CRUD interface for Trigger
      */
+
+    /**
+     * Create a new <code>Trigger</code>.  <code>Conditions</code> and <code>Actions</code> are manipulated in separate
+     * calls. The new </code>Trigger</code> will be persisted.  When fully defined a call to
+     * {@link #updateTrigger(Trigger)}
+     * is needed to enable the </code>Trigger</code>.
+     * @param trigger
+     * @throws Exception If the </code>Trigger</code> already exists.
+     */
     void addTrigger(Trigger trigger) throws Exception;
+
+    /**
+     * The <code>Trigger</code> will be removed from the Alerts engine, as needed, and will no longer be persisted.
+     * @param triggerId
+     * @throws Exception
+     */
     void removeTrigger(String triggerId) throws Exception;
-    void updateTrigger(Trigger trigger) throws Exception;
-    Collection<Trigger> getTriggers() throws Exception;
+
+    /**
+     * Update the <code>Trigger</code>. <code>Conditions</code> and <code>Actions</code> are manipulated in separate
+     * calls. The updated </code>Trigger</code> will be persisted.  If enabled the </code>Trigger</code>
+     * will be [re-]inserted into the Alerts engine and any prior dampening will be reset.
+     * @param triggerId
+     * @throws Exception If the </code>Trigger</code> does not exist.
+     */
+    Trigger updateTrigger(Trigger trigger) throws Exception;
+
     Trigger getTrigger(String triggerId) throws Exception;
 
+    Collection<Trigger> getAllTriggers() throws Exception;
```

Comment:
Disclaimer: I'm using this PR not only to review but also for brainstorming :-).
I am wondering if it is time to introduce a tags/groups feature in the model.
We have requeriments as https://issues.jboss.org/browse/HWKALERTS-12 where we should handle "groups/profiles" in some way.
Without enter in a very low level detail yet, I guess that this can help us to prototype some kind of tagging mechanism.
My main doubt/fear with grouping is that at the end we should not depend of the inventory model, I mean, our tagging/group solution should be valid with or without inventory in the equation.
Having said that, looking at the DefinitionsService I ask myself if we should have also a Collection<Trigger> getTriggersByTag(Set<Tag> tags) or similar to have a starting point.